### PR TITLE
Rest test fixes

### PR DESCRIPTION
**Pull request by clintongormley**
These are various fixes I had to make to get Perl running with the rest tests.

* Adjust percentile tests to work with Perl number handling
* Fix bad YAML in search/110_field_collapsing.yml 
* Force dummy scripts to be strings, not numbers 
* Fixed bad YAML comments
* Removed unneeded "headers" from bulk/10_basic.yml
* Rename remote.info to cluster.remote_info 
* Rename render_search_template to indices.render_search_template 
* Rename ingest.processor.grok to ingest.processor_grok

### CODE REVIEW

**Comment by karmi** [comment]
Thanks for looking into that, @clintongormley! I've run the Ruby test suite against the patch and I'm getting more failures than on current master :), I'll investigate more in the next couple of days.

File: rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/180_percentiles_tdigest_metric.yml [hunk 1]
```diff
@@ -62,27 +62,47 @@ setup:
...
-      aggregations.percentiles_double.values:
-        "1.0": 2.5
-        "5.0": 8.500000000000002
-        "25.0": 38.5
-        "50.0": 76.0
-        "75.0": 113.5
-        "95.0": 143.49999999999997
-        "99.0": 149.5
+
+  # - match:
```
Comment (karmi) [inline rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/180_percentiles_tdigest_metric.yml]: The fixes made by @fxdgear in #26905, and my subsequent fixes in #27355 worked nicely for me in Ruby. I must admit I like more the original notation. Also, some tests in the proposed notation (`aggregations.percentiles_int.values.5\.0`) fail for me, not sure how the Ruby runner handles it, must debug...

Comment (clintongormley) [reply rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/180_percentiles_tdigest_metric.yml]: The test runners are supposed to support dots escaped with `\`.

The original notation is a non-starter for me, for instance Perl gets back `143.5` and keeps it as that value, rather than converting to `143.49999999999997`.  But then it sees `1.0` in the response and reads that as `1`(then the test fails when it tries to compare it to `1.0`).

I can do numeric comparisons when there is a single numeric-like value on the right, but not with an object.

Comment (clintongormley) [reply rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/180_percentiles_tdigest_metric.yml]: What I could do (keeping the key-to-numeric-value format) is allow for approximate numeric matching (ie within a certain tolerance) so that I can remove the `lte`/`gte`

File: rest-api-spec/src/main/resources/rest-api-spec/api/indices.render_search_template.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "render_search_template": {
+  "indices.render_search_template": {
```
Comment (Mpdreamz) [inline rest-api-spec/src/main/resources/rest-api-spec/api/indices.render_search_template.json]: But nit picky perhaps but I think this actually makes more sense unnamespaced like other search related API's. Do not feel strongly about this though.

**Review by Mpdreamz — APPROVED** [review]
Other than a small nitpick LGTM

File: rest-api-spec/src/main/resources/rest-api-spec/api/indices.render_search_template.json (hunk 2 above)
Comment (clintongormley) [reply rest-api-spec/src/main/resources/rest-api-spec/api/indices.render_search_template.json]: Thanks, you're right - don't know what i was thinking

File: rest-api-spec/src/main/resources/rest-api-spec/test/bulk/10_basic.yml [hunk 3]
```diff
@@ -65,12 +65,9 @@
   - skip:
       version: " - 5.4.99"
       reason: confusing exception messaged caused by empty object fixed in 5.5.0
-      features: ["headers"]
 
   - do:
       catch: /Malformed action\/metadata line \[3\], expected FIELD_NAME but found \[END_OBJECT\]/
-      headers:
-        Content-Type: application/json
```
Comment (javanna) [inline rest-api-spec/src/main/resources/rest-api-spec/test/bulk/10_basic.yml]: why did you remove these? I'm afraid they were there for a reason. This may work for clients without the header as they always use json content-type anyway, but providing the header disables the content-type randomization that the java yaml test runner usually does.

File: modules/ingest-common/src/test/resources/rest-api-spec/test/ingest/120_grok.yml [hunk 4]
```diff
@@ -158,6 +158,6 @@ teardown:
 ---
 "Test Grok Patterns Retrieval":
   - do:
-      ingest.processor.grok: {}
+      ingest.processor_grok: {}
```
Comment (javanna) [inline modules/ingest-common/src/test/resources/rest-api-spec/test/ingest/120_grok.yml]: wondering if you were the first one ever running these tests...

Comment (javanna) [reply modules/ingest-common/src/test/resources/rest-api-spec/test/ingest/120_grok.yml]: I think I don't follow this change, why the underscore instead of the dot? the API is called `ingest.processor.grok` in the spec right? This should fail the build I think.

File: rest-api-spec/src/main/resources/rest-api-spec/test/bulk/10_basic.yml (hunk 3 above)
Comment (clintongormley) [reply rest-api-spec/src/main/resources/rest-api-spec/test/bulk/10_basic.yml]: OK - I can revert this and add a `features` skip clause instead.  The problem was that it had headers, but hadn't declared the need for the headers feature.

File: modules/ingest-common/src/test/resources/rest-api-spec/test/ingest/120_grok.yml (hunk 4 above)
Comment (clintongormley) [reply modules/ingest-common/src/test/resources/rest-api-spec/test/ingest/120_grok.yml]: I renamed `ingest.processor.grok` to `ingest.processor_grok` in the spec too, so this should work.  The reason for the underscore instead of the dot is that an underscore returns a new namespace in the REST clients, but we have only one thing in this namespace, so makes more sense just to have it at the same level as the other ingest methods.

File: rest-api-spec/src/main/resources/rest-api-spec/test/bulk/10_basic.yml (hunk 3 above)
Comment (clintongormley) [reply rest-api-spec/src/main/resources/rest-api-spec/test/bulk/10_basic.yml]: Ah, I see that the `headers` feature has already been added, but it's impossible to ignore without also adding parsing to skip it... which i shall duly do (and revert this change)

File: modules/ingest-common/src/test/resources/rest-api-spec/test/ingest/120_grok.yml (hunk 4 above)
Comment (javanna) [reply modules/ingest-common/src/test/resources/rest-api-spec/test/ingest/120_grok.yml]: Yes, I see that now. And the name of the api didn't match the name of the file. All good. I am going to open a PR to validate that the name of the api matches the filename, I've seen enough of these errors.

File: rest-api-spec/src/main/resources/rest-api-spec/test/bulk/10_basic.yml (hunk 3 above)
Comment (javanna) [reply rest-api-spec/src/main/resources/rest-api-spec/test/bulk/10_basic.yml]: you mean that you would expect two skip clauses here? or the reason also highlighting why the headers skip?

Comment (javanna) [reply rest-api-spec/src/main/resources/rest-api-spec/test/bulk/10_basic.yml]: ok sorry we were kind of typing at the same time, ignore my last message ;)

**Comment by clintongormley** [comment]
jenkins test this

File: rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/180_percentiles_tdigest_metric.yml (hunk 1 above)
Comment (polyfractal) [reply rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/180_percentiles_tdigest_metric.yml]: I'm thinking we may just want to remove the values like `143.49999999999997`.  I didn't think about it at the time when I added the test, but these are likely subject to how each language interprets json floats and internal float vs double, etc.

Would simplify everyone's life if we stuck to nice "discrete" floats (`149.5`, etc).  The point of these REST tests are to just do rough validation of the body, not really the values, so I think it's fine if we omit the problematic percentiles.

File: rest-api-spec/src/main/resources/rest-api-spec/test/bulk/10_basic.yml (hunk 3 above)
Comment (karmi) [reply rest-api-spec/src/main/resources/rest-api-spec/test/bulk/10_basic.yml]: I think I've fixed this in https://github.com/elastic/elasticsearch/pull/26896, but I might have forgot to cherry-pick that commit into the `6.0` branch, which [I did yesterday](https://github.com/elastic/elasticsearch/commit/12a128beb7f51151b852be38bc52e3107a202a82).

**Comment by clintongormley** [comment]
thanks @Mpdreamz @javanna @karmi for the reviews.  merging
